packages/assets-controllers/src/NftDetectionController.test.ts

```diff
@@ -378,13 +480,107 @@ describe('NftDetectionController', () => {
+        await controller.start();
+        // await clock.tickAsync(pollingInterval);
+
+        expect(mockAddNft).toHaveBeenNthCalledWith(
+          1,
+          '0xCE7ec4B2DfB30eB6c0BB5656D33aAd6BFb4001Fc',
+          '2577',
+          {
+            nftMetadata: {
+              description:
+                "Redacted Remilio Babies is a collection of 10,000 neochibi pfpNFT's expanding the Milady Maker paradigm with the introduction of young J.I.T. energy and schizophrenic reactionary aesthetics. We are #REMILIONAIREs.",
+              image: 'https://imgtest',
+              imageOriginal: 'https://remilio.org/remilio/632.png',
+              imageThumbnail: 'https://imgSmall',
+              name: 'Remilio 632',
+              rarityRank: 8872,
+              rarityScore: 343.443,
+              standard: 'ERC721',
+            },
+            userAddress: '0x1',
+            source: Source.Detected,
+          },
+        );
+        expect(mockAddNft).toHaveBeenNthCalledWith(
+          2,
+          '0x0B0fa4fF58D28A88d63235bd0756EDca69e49e6d',
+          '2578',
+          {
+            nftMetadata: {
+              description: 'Description 2578',
+              image: 'https://imgtest',
+              imageOriginal: 'https://remilio.org/remilio/632.png',
+              imageThumbnail: 'https://imgSmall',
+              name: 'ID 2578',
+              rarityRank: 8872,
+              rarityScore: 343.443,
+              standard: 'ERC721',
+            },
+            userAddress: '0x1',
+            source: Source.Detected,
+          },
+        );
+        expect(mockAddNft).toHaveBeenNthCalledWith(
+          3,
+          '0xebE4e5E773AFD2bAc25De0cFafa084CFb3cBf1eD',
+          '2574',
+          {
+            nftMetadata: {
+              description: 'Description 2574',
+              image: 'image/2574.png',
+              imageOriginal: 'imageOriginal/2574.png',
+              name: 'ID 2574',
+              standard: 'ERC721',
+            },
+            userAddress: '0x1',
+            source: Source.Detected,
+          },
+        );
+
+        controllerEvents.networkStateChange({
+          ...defaultNetworkState,
+          selectedNetworkClientId: 'AAAA-AAAA-AAAA-AAAA',
+        });
+        await clock.tickAsync(pollingInterval);
+
+        // Not 6 times, which is what would happen if detectNfts were called
+        // again
+        expect(mockAddNft).toHaveBeenCalledTimes(3);
+      },
+    );
+  });
+
   it('should detect and add NFTs correctly when blockaid result is not included in response', async () => {
     const mockAddNft = jest.fn();
     await withController(
       { options: { addNft: mockAddNft } },
-      async ({ controller, triggerPreferencesStateChange }) => {
+      async ({ controller, controllerEvents }) => {
         const selectedAddress = '0x1';
-        triggerPreferencesStateChange({
+        controllerEvents.preferencesStateChange({
```

Comment:
Note that there are a lot of changes in this PR that are unrelated to the change from `providerConfig` to `selectedNetworkClientId`. I took this opportunity to align the tests in this file to use our latest convention for mocking events couched as callbacks.